common-custom-user-data-gradle-plugin/src/main/java/com/gradle/SystemPropertyOverrides.java

```diff
@@ -46,6 +51,9 @@ static void configureBuildCache(BuildCacheConfiguration buildCache, ProviderFact
         if (buildCache.getRemote() != null) {
             buildCache.remote(HttpBuildCache.class, remote -> {
                 sysProperty(REMOTE_CACHE_URL, providers).ifPresent(remote::setUrl);
+                if (remote.getUrl() != null) {
```

Comment:
I'm not sure if this check is strictly needed because it seems unlikely that a user would have a HttpBuildCache configured with a null URL.